packages/typegen/src/generate/events.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017-2020 @polkadot/typegen authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import Handlebars from 'handlebars';
+
+import staticData from '@polkadot/metadata/static';
+import { Metadata } from '@polkadot/metadata';
+import { TypeRegistry } from '@polkadot/types/create';
+import * as defaultDefs from '@polkadot/types/interfaces/definitions';
+import { stringCamelCase } from '@polkadot/util';
+
+import { createImports, compareName, formatType, readTemplate, registerDefinitions, setImports, writeFile } from '../util';
+
+const template = readTemplate('events');
+const generateForMetaTemplate = Handlebars.compile(template);
+
+/** @internal */
+function generateForMeta (meta: Metadata, dest: string, extraTypes: Record<string, Record<string, { types: Record<string, any> }>>, isStrict: boolean): void {
+  writeFile(dest, (): string => {
+    const allTypes: Record<string, Record<string, { types: Record<string, unknown> }>> = { '@polkadot/types/interfaces': defaultDefs, ...extraTypes };
+    const imports = createImports(allTypes);
+    const allDefs = Object.entries(allTypes).reduce((defs, [path, obj]) => {
+      return Object.entries(obj).reduce((defs, [key, value]) => ({ ...defs, [`${path}/${key}`]: value }), defs);
+    }, {});
+
+    const modules = meta.asLatest.modules
+      .sort(compareName)
+      .filter((mod) => mod.events.isSome)
+      .map((mod) => {
+        return { ...mod, events: mod.events.unwrap() };
+      })
+      .filter((mod) => mod.events.length > 0)
+      .map(({ events, name }) => {
+
+        const items = events
+          .sort(compareName)
+          .map(({ args, documentation, name }) => {
+            const argTypes = args.map((type) => formatType(allDefs, type.toString(), imports))
+            setImports(allDefs, imports, argTypes);
+
+            return {
+              docs: documentation,
+              name: stringCamelCase(name),
```

Comment:
Since there are types-only, I would keep it to non-camelCase, i.e. most TS types are generally `SomeType` vs `someType`.